zenodo/modules/records/templates/zenodo_records/box/info.html

```diff
@@ -96,22 +96,24 @@
   {%- if i and not p %}
     <dt>{{_('Imprint')}}:</dt>
     <dd>
-      {{i.publisher}}{% if i.place %}, {{i.place}}{%- endif %}{% if i.isbn %} ({{i.isbn}}){%- endif %}.
+      {{i.publisher}}{% if i.place %}, {{i.place}}{%- endif %}{% if i.isbn %}({{i.isbn}}){%- endif %}.
     </dd>
-  {%- elif p %}
+  {%- elif p and p.title %}
     <dt>{{_('Published in')}}:</dt>
     <dd>
-      {{p.title}}{% if i.place %}, {{i.publisher}}{%- endif %}{% if i.place %}, {{i.place}}{%- endif %}{% if p.pages %}, pp. {{p.pages}}{%- endif %}{% if i.isbn %} ({{i.isbn}}){%- endif %}.
+        {{p.title}}{% if i.publisher %}, {{i.publisher}}{%- endif %}{% if i.place %}, {{i.place}}{%- endif %}{% if p.pages %}, pp. {{p.pages}}{%- endif %}{% if i.isbn %}({{i.isbn}}){%- endif %}.
```

Comment:
Hey, looks good, one last thing - 
You removed a space after before the (ISBN): ``{% if i.isbn %} ({{i.isbn}}){%- endif %}``, please bring it back and we can merge it :)